XenAdminTests/ArchiveTests/ArchiveIteratorTests.cs

```diff
@@ -152,96 +147,101 @@ public void TearDown()
         [SetUp]
         public void TestSetup()
         {
+            tempPath = null;
             fakeIterator.Reset();
         }
 
+        [TearDown]
+        public void TestTearDown()
+        {
+            if (Directory.Exists(tempPath))
+                Directory.Delete(tempPath, true);
+        }
+
 
         [Test]
-        public void AnExceptionIsThrownForNullArgumentWhenCallingExtractAllContents()
+        public void TestExtractToNullDestinationPath()
         {
             Assert.Throws(typeof(ArgumentNullException), () => fakeIterator.ExtractAllContents(null));
         }
 
         [Test]
-        public void AnExceptionIsThrownForANullFileNameWhenCallingExtractAllContents()
+        public void TestExtractNullFile()
         {
             fakeIterator.CurrentFileNameReturn = null;
-            Assert.Throws(typeof(ArgumentNullException), () => fakeIterator.ExtractAllContents(Path.GetTempPath()));
+            Assert.Throws(typeof(NullReferenceException), () => fakeIterator.ExtractAllContents(Path.GetTempPath()));
         }
-        
-        [Test]
-        public void VerifyAFileIsWrittenWhenCallingExtractAllContents()
+
+        [TestCase(true, true)]
+        [TestCase(true, false)]
+        [TestCase(false, true)]
+        [TestCase(false, false)]
+        public void TestExtractFile(bool longDirectoryPath, bool longFilePath)
         {
-            string tempPath = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+            tempPath = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+            
+            var dirCharNumber = (longDirectoryPath ? 248 : 247) - tempPath.Length - 2;

```

Comment:
I've changed my mind several times about this, eventually I thought that a util method is more confusing because of the additional number of characters removed and their justification which depends on the test case. Hence I put the whole calculation in the test.